maro/simulator/scenarios/data_center/business_engine.py

```diff
@@ -0,0 +1,334 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+from typing import Dict, List
+
+from yaml import safe_load
+
+from maro.event_buffer import AtomEvent, CascadeEvent, EventBuffer, MaroEvents
+from maro.simulator.scenarios.abs_business_engine import AbsBusinessEngine
+from maro.simulator.scenarios.helpers import DocableDict
+from maro.utils.utils import DottableDict
+
+from .common import (
+    AssignAction, DecisionPayload, Latency, PostponeAction, PostponeType, ValidPhysicalMachine, VmFinishedPayload,
+    VmRequirementPayload
+)
+from .events import Events
+from .physical_machine import PhysicalMachine
+from .virtual_machine import VirtualMachine
+
+metrics_desc = """
+
+total_energy_consumption (int): Current total energy consumption.
+success_allocation (int): Accumulative successful VM allocation until now.
+success_placement (int): Accumulative successful VM placement.
+failed_placement (int): Accumulative failed VM placement until now.
+total_latency (int): Accumulative used buffer time until now.
+"""
+
+
+class DataCenterBusinessEngine(AbsBusinessEngine):
```

Comment:
The naming ``data center`` may be too big for the problem we focused on currently.

How about changing to ``VM Allocation`` or ``VM Scheduling``?
BTW, since both these 2 namings are used in this domain, which one do you prefer?

Reply:
How about VM placement?

Reply:
Since "vm allocation or vm scheduling" means only allocate or schedule the VMs. But we have considered more conditions, like ober-subscription, energy, etc.

Reply:
Some reference that mention VM placement:
1. https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=9139719
2. https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=7750592